transform/convert_quadkey_and_Vertical_id_test.go

```diff
@@ -597,3 +689,83 @@ func TestSpatialIDCheckZoom(t *testing.T) {
 	}
 
 }
+
+// expected output is defined as outputIndex, e_i, where e_i is the set of integers that satisfy the solution set conditions:
+//  1. the altitudes of the input index is:
+//     (inputIndex)*(2^25-inputZoom) <= altitude < (inputIndex +1)*(2^25-inputZoom)
+//  2. the altitudes associated with index i is:
+//     (index_i + indexOffset)*(2^25-outputZoom) <= altitude < (index_i + indexOffset +1)*(2^25-outputZoom)
+//  3. the minimum altitude of index i is less than the maximum altitude of the inputIndex
+//  4. assuming inputIndex exists, the number of integers in e_i >= 1
+func TestConvertVerticalIndex(t *testing.T) {
+	datas := []struct {
+		inputZoom           int64
+		inputIndex          int64
+		outputZoom          int64
+		indexOffset         int64
+		altitudeRangeScalar int64
+		expectedOutputIndex []int64
+	}{
+		{inputZoom: 25, outputZoom: 27, inputIndex: 100, indexOffset: 0, expectedOutputIndex: []int64{400, 401, 402, 403}},
+		{inputZoom: 25, outputZoom: 24, inputIndex: 100, indexOffset: 0, expectedOutputIndex: []int64{50}},
+		{inputZoom: 25, outputZoom: 25, inputIndex: 100, indexOffset: 0, expectedOutputIndex: []int64{100}},
+		{inputZoom: 25, outputZoom: 25, inputIndex: 100, indexOffset: -47, expectedOutputIndex: []int64{53}},
+		{inputZoom: 25, outputZoom: 26, inputIndex: 0, indexOffset: 3, expectedOutputIndex: []int64{3, 4}},
+		{inputZoom: 25, outputZoom: 21, inputIndex: 100, indexOffset: -17, expectedOutputIndex: []int64{-11}},
+		{inputZoom: 25, outputZoom: 25, inputIndex: 100, indexOffset: -512, expectedOutputIndex: []int64{-412}},
+		{inputZoom: 25, outputZoom: 14, inputIndex: 28, indexOffset: 1000, expectedOutputIndex: []int64{1000}},
+		{inputZoom: 25, outputZoom: 24, inputIndex: 100, altitudeRangeScalar: 1, indexOffset: 0, expectedOutputIndex: []int64{100}},
+		{inputZoom: 25, outputZoom: 21, inputIndex: 100, altitudeRangeScalar: 5, indexOffset: -17, expectedOutputIndex: []int64{183, 184}},
+		{inputZoom: 25, outputZoom: 27, inputIndex: 100, altitudeRangeScalar: 1, indexOffset: 0, expectedOutputIndex: []int64{800}},
+	}
+
+	for _, p := range datas {
+		result, error := convertVerticalIndex(p.inputIndex, p.inputZoom, p.outputZoom, p.altitudeRangeScalar, p.indexOffset)
+		if error != nil {
+			t.Log(t.Name())
+			t.Error(error)
+		}
+		for i := 0; i < len(p.expectedOutputIndex); i++ {

```

Comment:
This is wrong in the case where len(p.expectedOutpuIndex) is less than len(result).
Indeed, the datas[10] is not checked enough.
Append length check.